riscv/processor.cc

```diff
@@ -797,9 +812,13 @@ void processor_t::disasm(insn_t insn)
       fprintf(log_file, "core %3d: Executed %" PRIx64 " times\n", id, executions);
     }
 
-    fprintf(log_file, "core %3d: 0x%0*" PRIx64 " (0x%08" PRIx64 ") %s\n",
-            id, max_xlen/4, zext(state.pc, max_xlen), bits,
-            disassembler->disassemble(insn).c_str());
+    stringstream s;  // first put everything in a string, later send it to output
+    s << "core " << dec << setfill(' ') << setw(3) << id
+      << hex << ": 0x" << setfill('0') << setw(max_xlen/4)
+      << zext(state.pc, max_xlen) << " (0x" << setw(8) << bits << ") "
+      << disassembler->disassemble(insn) << endl;
+    if (log_file==stderr) *sout_ptr << s.str(); // handles command line options -d -s -l
+    else fputs(s.str().c_str(),log_file); // handles command line option --log
```

Comment:
Could we encapsulate this unusual if/else behavior in a new function (and call the same function in the previous case)?

Also, please separate operators and function args by spaces, and in most cases put the body on a new line, indented.  e.g.:

```
    if (log_file == stderr)
      *sout_ptr << s.str(); // handles command line options -d -s -l
    else
      fputs(s.str().c_str(), log_file); // handles command line option --log
```